### PR TITLE
fix: allow props to change the inline spinner again

### PR DESCRIPTION
**Pull request by snapsnapturtle**
**What:**
Re-enable all props for the InlineSpinner component and fixed the tests, so we don't break it again 👀 
​
**Why:**
We introduced a wrapper element for the InlineSpinner in #77, but didn't forward the props for margin and size. This resulted in typescript projects that set the margin to break. All other projects would simply lose their set margins and sizes. @nlopin already found the issue with the sizes in #95, which this PR closes as well.
​
**How:**
Instead of wrapping the `div` element with `span`, we set the original element to `span`.
​
**Checklist:**
- [x] Ready to be merged

### CODE REVIEW

**Comment by nlopin** [comment]
Have you checked the new approach works in the [example Alexis reported](https://codesandbox.io/s/button-spinner-issue-wnmgv)?

**Comment by snapsnapturtle** [comment]
Yes @nlopin, I did check it with the example. I was hoping to be able to write a test for it, but without visual comparisons it's very difficult to have this testable

File: src/components/InlineSpinner/__snapshots__/InlineSpinner.spec.tsx.snap [hunk 1]
```diff
@@ -7,16 +7,20 @@ exports[`InlineSpinner renders in a custom color 1`] = `
   width: 1.25rem;
   height: 1.25rem;
   vertical-align: text-bottom;
-  border: 0.125rem solid;
+  border: 0.125rem solid #069D4F;
```
Comment (nlopin) [inline src/components/InlineSpinner/__snapshots__/InlineSpinner.spec.tsx.snap:10]: from where this color comes from? shouldn't it be `inherit`?

**Review by nlopin — APPROVED** [review]

File: src/components/InlineSpinner/__snapshots__/InlineSpinner.spec.tsx.snap (hunk 1 above)
Comment (snapsnapturtle) [reply src/components/InlineSpinner/__snapshots__/InlineSpinner.spec.tsx.snap:10]: This is the test to ensure the spinner accepts a custom color 🎨

**Pull request merged**